src/clr-angular/forms/common/providers/control.service.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable, OnDestroy} from "@angular/core";
+import {NgControl} from "@angular/forms";
+import {Observable, Subject, Subscription} from "rxjs";
+
+@Injectable()
+export class ControlService implements OnDestroy {
+    // Implement our own since it doesn't fire on events like blur, and return the control instead of a string
+    private _statusChanges: Subject<NgControl> = new Subject();
+    get statusChanges(): Observable<NgControl> {
+        return this._statusChanges.asObservable();
+    }
+
+    private subscription: Subscription;
+
+    //
+    private _control: NgControl;
+    get control(): NgControl {
+        return this._control;
+    }
+    set control(control: NgControl) {
+        this._control = control;
+        this.subscribe();
+    }
+
+    // When the control is set, we want to subscribe to statusChanges and forward them to
+    // our observable when the control is invalid (dirty,touched)
+    private subscribe() {
+        if (this.subscription) {
+            this.subscription.unsubscribe();
+        }
+
+        if (this._control.statusChanges) {
+            this.subscription = this._control.statusChanges.subscribe(status => {
+                if (this._control.dirty && this._control.touched) {
```

Comment:
I thought we said touched was enough for required fields to throw an error? Based on what I'm reading here, if I focus in then out of a required input, I do not get the error message.

I think this should be `if (this._control.dirty || this._control.touched)`, but maybe I missed something.

Reply:
It doesn't really matter if the control is dirty or not, if it is touched is when we care to check it. This makes sense when a form is new or when the data loaded is valid, but might not make sense if the control could preload invalid data (not sure when this could be the case, but perhaps it could be, perhaps when a form is autosaved even if its incomplete). 

Perhaps this is something that can/should be configurable?

Reply:
Invalid pre-loaded data has to come from the user anyway, right? So without Clarity, it seems logical to me that the way to do this in Angular is to load the previously entered user data on initialization, and to mark the corresponding fields as dirty since they were "entered" by the user. 
So I think Clarity would just do the same, we don't have to provide anything, they just get hold the the FormControls (which isn't even extra work with reactive forms) and mark the ones the pre-filled with user data as dirty.